File: front/src/types/usinas.ts
export interface UsinaOut {
  usina_id: string
  nome: string
  fonte: "solar" | "eolica"
  potencia_mw: number
  submercado: string
  latitude?: number | null
  longitude?: number | null
  id_ons?: string | null
  ceg?: string | null
  nom_conjuntousina?: string | null
  nom_usina_conjunto?: string | null
  total_corte_mwh?: number | null
  total_perda_reais?: number | null
  total_intervalos_restricao?: number | null
  data_inicio?: string | null
  data_fim?: string | null
}

export interface UsinaListOut {
  total_count: number
  limit: number
  offset: number
  items: UsinaOut[]
}


export interface RiscoHora {
  timestamp: string
  probabilidade_corte: number
  magnitude_estimada_mwh: number
}

export interface UsinaResumoOut {
  usina: UsinaOut
  total_corte_mwh: number
  total_perda_reais: number
  percentual_ressarcivel: number
  perda_ressarcivel_reais: number
  perda_por_razao: Record<string, number>
  total_eventos_corte: number
  total_intervalos_restricao?: number
  ticket_medio_evento_reais: number
  perda_esperada_30d: {
    valor_reais: number
    horizonte_dias: number
    metodo: string
    observacao?: string | null
  }
}

File: backend/app/schemas/usinas.py
from pydantic import BaseModel

from app.schemas.common import MetaOut


class UsinaOut(BaseModel):
    usina_id: str
    nome: str
    fonte: str
    potencia_mw: float
    submercado: str
    latitude: float | None = None
    longitude: float | None = None
    id_ons: str | None = None
    ceg: str | None = None
    nom_conjuntousina: str | None = None
    nom_usina_conjunto: str | None = None
    total_corte_mwh: float | None = None
    total_ressarcivel_mwh: float | None = None
    total_perda_reais: float | None = None
    total_perda_ressarcivel_reais: float | None = None
    pld_faltante_intervalos: int | None = None
    total_intervalos_restricao: int | None = None
    data_inicio: str | None = None
    data_fim: str | None = None


class UsinaListOut(BaseModel):
    total_count: int
    limit: int
    offset: int
    metadata: MetaOut
    items: list[UsinaOut]


class PerdaEsperadaOut(BaseModel):
    valor_reais: float
    horizonte_dias: int
    metodo: str
    observacao: str | None = None


class UsinaResumoOut(BaseModel):
    usina: UsinaOut
    total_corte_mwh: float
    total_perda_reais: float
    percentual_ressarcivel: float
    perda_ressarcivel_reais: float
    perda_por_razao: dict[str, float]
    total_eventos_corte: int
    total_intervalos_restricao: int = 0
    ticket_medio_evento_reais: float
    metadata: MetaOut
    perda_esperada_30d: PerdaEsperadaOut
